src/time_service/index.ts

```diff
@@ -2,12 +2,30 @@ import fetch from 'node-fetch';
 import { config } from '../time_service/config';
 
 interface TimeService {
-    getTime(date: Date): Promise<number>;
+    getTimeFromServer(date: Date): Promise<number>;
+    getTime();
 }
 
 // TODO calc delivery time for request
-export class TimeServiceClient implements TimeService {
-    async getTime(): Promise<number> {
+class TimeServiceClient implements TimeService {
+
+    private timeDifference: number;
+
+    constructor() {
+        this.timeDifference = 0;
+        this.init();
```

Comment:
you are calling async function here. so constructor will return Promise, not instance.